api_app/script_analyzers/observable_analyzers/inquest.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import requests
+
+from api_app.exceptions import AnalyzerRunException, AnalyzerConfigurationException
+from api_app.script_analyzers.classes import ObservableAnalyzer
+from intel_owl import secrets
+
+
+logger = logging.getLogger(__name__)
+
+
+class InQuest(ObservableAnalyzer):
+    base_url: str = "https://labs.inquest.net"
+
+    def set_config(self, additional_config_params):
+        self.api_key_name = additional_config_params.get(
+            "api_key_name", "INQUEST_API_KEY"
+        )
+        self.__api_key = secrets.get_secret(self.api_key_name)
+        self.analysis_type = additional_config_params.get(
+            "inquest_analysis", "dfi_search"
+        )
+
+    @property
+    def hash_type(self):
+        hash_lengths = {32: "md5", 40: "sha1", 64: "sha256", 128: "sha512"}
+        hash_type = hash_lengths.get(len(self.observable_name), None)
+        if not hash_type:
+            raise AnalyzerConfigurationException(
+                f"Given Hash: '{hash}' is not suported."
+                "Supported hash types are: 'md5', 'sha1', 'sha256', 'sha512'."
+            )
+        return hash_type
+
+    def run(self):
+        result = {}
+        headers = {"Content-Type": "application/json"}
+        if self.__api_key:
+            headers["Authorization"] = self.__api_key
+        else:
+            warning = f"No API key retrieved with name: {self.api_key_name}"
+            logger.info(
+                f"{warning}. Continuing without API key..." f" <- {self.__repr__()}"
+            )
+            self.report["errors"].append(warning)
+
+        if self.analysis_type == "dfi_search":
+            if self.observable_classification == "hash":
+                uri = "/api/dfi/search/hash/{0}?hash={1}".format(
```

Comment:
not a real problem, but I would ask you to avoid to use f-strings and format strings in the same code. This may create headache to developers. Please follow f-format strings everywhere

Reply:
+1